src/libsync/propagatedownload.h

```diff
@@ -110,6 +112,10 @@ class OWNCLOUDSYNC_EXPORT GETFileJob : public AbstractNetworkJob
     qint64 expectedContentLength() const { return _expectedContentLength; }
     void setExpectedContentLength(qint64 size) { _expectedContentLength = size; }
 
+protected:
+    virtual qint64 writeToDevice(const char *data, qint64 len);
```

Comment:
please use `QByteArray` instead of `const char *` + `qint64`

Reply:
@mgallien Somehow, after this change, the feature got broken. Will need to investigate why it doesn't work with `QByteArray`. Could be, due to a zero symbol at the end or something...

Reply:
@mgallien So, the issue is with `QByteArray` not storing all the data if there is a '\0' symbol encountered that is considered as an end of a string. I am using `QByteArray::fromRaw` to make the code work after I've implemented your comment. Though, I don't really like this change, because, `_device->write(data, size)` was there before and looks like it was done for a reason.

Reply:
so should this be set resolved ?
looks to me like you are using QByteArray and you fixed the bug